work_test.go

```diff
@@ -228,31 +219,33 @@ func TestLockJobAdvisoryRace(t *testing.T) {
 	}
 
 	// helper functions
-	newConn := func() *pgx.Conn {
-		conn, err := pgx.Connect(testConnConfig)
+	newConn := func() *sql.Conn {
+		db, err := sql.Open("pgx", testConnConfig)
+		if err != nil {
+			panic(err)
+		}
+		conn, err := db.Conn(context.Background())
 		if err != nil {
 			panic(err)
 		}
 		return conn
 	}
-	getBackendID := func(conn *pgx.Conn) int32 {
+	getBackendID := func(conn *sql.Conn) int32 {
 		var backendID int32
-		err := conn.QueryRow(`
-			SELECT backendid
-			FROM pg_stat_get_backend_idset() psgb(backendid)
-			WHERE pg_stat_get_backend_pid(psgb.backendid) = pg_backend_pid()
+		err := conn.QueryRowContext(context.Background(), `
+			SELECT pg_backend_pid()
 		`).Scan(&backendID)
 		if err != nil {
 			panic(err)
 		}
 		return backendID
 	}
-	waitUntilBackendIsWaiting := func (backendID int32, name string) {
+	waitUntilBackendIsWaiting := func(backendID int32, name string) {
 		conn := newConn()
 		i := 0
 		for {
 			var waiting bool
-			err := conn.QueryRow(`SELECT pg_stat_get_backend_waiting($1)`, backendID).Scan(&waiting)
+			err := conn.QueryRowContext(context.Background(), `SELECT waiting from pg_stat_activity where pid=$1`, backendID).Scan(&waiting)
```

Comment:
I am using postgresql 9.6 and `pg_stat_get_backend_waiting` doesn't seem to exist in 9.6.